### ISSUE TITLE
Document how to `add` a dependency with an exact version

### ISSUE THREAD

**Issue by homeworkprod** [python-poetry/poetry#1440]
Hi, Poetry newbie here, using Poetry v0.12.17.

Trying to get my dependencies specified in `requirements.txt` format into Poetry took me longer than expected. Here's what I've found.

- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.

The short documentation section on [Exact [version] requirements](https://poetry.eustace.io/docs/versions/#exact-requirements), as opposed to others, includes no examples of the specific version format string and, this feels especially important for this case, a full Poetry command to add a specific version of a dependency.

I wanted to install a specific version of the `attrs` package (as an example), so I tried this:

- `poetry add attrs 19.1.0` results in
  ```
  Using version ^19.2 for attrs
  
  [PackageNotFound]  
  Package [19.1.0] not found.
  ```
  Nothing added to `pyproject.toml`.
  It seems that Poetry expects the version number to be a package name, which fits the usage string of the `add` command (`add [options here …] <name> (<name>)...`). That is fine. 

- `poetry add attrs==19.1.0`
  This is the format I expect to work as I'm used to it from `requirements.txt` and probably `setup.py` and other common, but older, approaches to specify requirements.

  The line `attrs = "=19.1.0"` gets added to `pyproject.toml`. This doesn't look wrong per se, but my understanding is that the equality sign is at least redundant/superfluous here and, thus, should not have been added automatically.

  And I'm not sure if it will actually work later on. Maybe Poetry assumes the second `=` is the start of the version number? That sounds like trouble.

  Maybe Poetry should normalize `==` to `=` as a special case?

- `poetry add attrs=19.1.0`
  This, to me, uses an unexpected format, coming from `requirements.txt` (where `==` seems to be/is common, though I'm not sure if a single `=` would work, too).

  However, the line `attrs = "19.1.0"` gets added to `pyproject.toml` and it looks correct to me.

- `poetry add 'attrs 19.1.0'`
  Okay, let's try to group package name and version number into arguments using quotes.

  Again, the line `attrs = "19.1.0"` gets added to `pyproject.toml`.

  This is fine, and not having to type `=` is nice (though the quotes then become necessary with this format), but should be documented.

- `poetry add 'attrs == 19.1.0'`
  This, with spaces around the `==`, is what's in my `requirements.txt` files, because it it more readable than without spaces, and works perfectly fine there (even though it's not the output generated by, for example, `pip freeze`, unfortunately).

  Result:
  ```
  [ValueError]               
  too many values to unpack (expected 2)
  ```
  Unexpected.

  As I've learned from `attrs==19.1.0` (with out without quotes around the term, btw), this doesn't seem to be what Poetry expects.

  However, I think it should be better at dealing with this, preferably even suggesting (or doing?) the right thing.

Note: I have removed any `attrs` dependency lines from `pyproject.toml` before executing each `poetry add` command in order to get independent results for the cases I described above.

**tl;dr**

The documentation should make clear how to add a dependency with a specific version using `poetry add`.

And maybe a dependency version value like `=1.2.3` ending up in `pyproject.toml` (both manually and via Poetry commands) should be caught and and error be raised.

Thanks for reading.

**Comment by homeworkprod** [comment]
I still see the need for an improvement, but I'm afraid I can't bring this forward myself at this point due to other active projects.

**Comment by maffka123** [comment]
Same here. I am trying to add [this](https://celonis.github.io/pycelonis/) package, and get access forbidden every time. With pip it works:
`pip install --extra-index-url=https://pypi.celonis.cloud/ pycelonis`

**Comment by AIGeneratedUsername** [comment]
This issue is still actual for Poetry 1.1.

**Comment by andrewasheridan** [comment]
`poetry add attrs@==19.1.0`

will add

`attrs = “==19.1.0”`


To the toml


and 

`poetry add attrs@19.1.0`

will add

`attrs = “19.1.0”`

to the toml 


Both of which appear to create the same lock file.

I don’t see this in the docs. 
- https://python-poetry.org/docs/dependency-specification/#exact-requirements

**Comment by Secrus** [comment]
This is now described in detail in docs.
